readme.md

```diff
@@ -1,64 +1,58 @@
 
->The problem with object-oriented languages is they’ve got all this implicit environment that they carry around with them. You wanted a banana but what you got was a gorilla holding the banana and the entire jungle. — Joe Armstrong
+When [Brendan Eich](http://blog.salsitasoft.com/why-now/) created Javascript for his then-employer Netscape, he was ordered by management to make the language look like Java. He obliged somewhat and gave Javascript some key functional programming features. As the language exploded in popularity, Javascript developers are now taking advantage of these features.
 
->I’ll never forget that day when I was ready to cash in on the promise of Reuse by inheriting from an existing class. This was the moment I had been waiting for.
-A new project came along and I thought back to that Class that I was so fond of in my last project.
-No problem. Reuse to the rescue. All I gotta do is simply grab that Class from the other project and use it.
-Well… actually… not just that Class. We’re gonna need the parent Class. But… But that’s it.
-Ugh… Wait… Looks like we gonna also need the parent’s parent too... And then… We’re going to need ALL of the parents. Okay… Okay… I handle this. No problem.
-And great. Now it won’t compile. Why?? Oh, I see… This object contains this other object. So I’m gonna need that too. No problem.
-Wait… I don’t just need that object. I need the object’s parent and its parent’s parent and so on and so on with every contained object and ALL the parents of what those contain along with their parent’s, parent’s, parent’s…
+## What is Functional Programming? (15 minutes)
 
-- https://medium.com/@cscalfani/goodbye-object-oriented-programming-a59cda4c0e53#.rmqoman2q
+Functional programming is characterized by **pure functions** and **function composition** and avoiding:
+* shared state
+* mutable data
+* side-effects
 
-Let's look at another programming paradigm, __functional programming__.
+***Huh?***
 
-Functional programming is characterized by:
+Trying to understand the terminology associated with functional programming can be incredibly daunting. In the scope of this class, we're not going to go too in the weeds with the concepts, but we'll have initial exposure.
 
-- Pure functions without side effects
-- Avoiding changing state and mutable data
+**Pure Functions**
 
-## Pure Functions (stateless)
+Pure functions are a fundamental part of functional programming.
 
->In mathematics, a function is a relation between a set of inputs and a set of permissible outputs with the property that each input is related to exactly one output. 
+When we say __pure__ we mean a function, given the same inputs, will always return the same output. Such a function **does not** rely on or modify the **state** of variables outside it's scope.
 
-<https://en.wikipedia.org/wiki/Function_(mathematics)>
+The execution of a pure function doesn't depend on the state of the system. That is, avoiding **shared state**.
```

Comment:
OCD Adrian here, but "That is, avoiding shared state" doesn't feel like a sentence.